wetterdienst/provider/dwd/mosmix/api.py

```diff
@@ -497,19 +497,15 @@ def _all(self) -> pd.DataFrame:
 
         df = pd.read_fwf(
             StringIO(payload.read().decode(encoding="latin-1")),
-            skiprows=4,
+            skiprows=2,
             skip_blank_lines=True,
             colspecs=[
                 (0, 5),
-                (6, 11),
-                (12, 17),
-                (18, 22),
-                (23, 44),
-                (44, 51),
-                (51, 58),
-                (59, 64),
-                (65, 71),
-                (72, 76),
+                (6, 10),
+                (11, 31),
+                (32, 38),
+                (39, 46),
+                (47, 52),
             ],
```

Comment:
I think this style of reading the file is prone to break again. Let's come up with a different solution.

Reply:
Look, there now is an indicator on the second line that indicates the colspecs/colwidth:

```
ID    ICAO NAME                 LAT    LON     ELEV
----- ---- -------------------- -----  ------- -----
```

How about parsing this line and deriving the colspecs from it?
https://www.dwd.de/EN/ourservices/met_application_mosmix/mosmix_stations.cfg;jsessionid=74F6A991EA8B280D4587EF6952CC77E3.live11043?view=nasPublication&nn=495490

Reply:
It looks like the machinery can infer the column specifications on its own, see eddc5ec09f6.

Reply:
Even better!

Reply:
Hi again,

with 76dc39b26d, I've re-added the improvement to infer the column specifications from the data itself, after learning from @aplavin, @gfyoung, and @aldanor at https://github.com/pandas-dev/pandas/issues/21970, that the number of rows to analyze should well be increased higher than the default of only 100 rows -- thanks a stack!

Apparently, the `infer_nrows` argument has been added to `read_fwf` for a while, so using `read_fwf(colspecs="infer", infer_nrows=100000)` did the trick for us.

With kind regards,
Andreas.